test/inductor/test_padded_tensor.py

```diff
@@ -0,0 +1,222 @@
+
+        f = torch.compile(f, fullgraph=True)
+        multipliers = {0: 16, 1: 16}
+
+        for i in range(3, 9):
+            # Set error_on_recompile to True after 3rd iteration.
+            # TODO: We don't need this if we mark padded dimensions as dynamic.
+            if i == 5:
+                torch._dynamo.config.error_on_recompile = True
+
+            a = PaddedTensor.from_tensor(torch.randn([3, 5]), multipliers)
+            b = PaddedTensor.from_tensor(torch.randn([5, i]), multipliers)
+            c = PaddedTensor.from_tensor(torch.randn([3, i]), multipliers)
+
+            y = f(a, b, c)
+
+            self.assertEqual(y.shape, (16, 16))
+            self.assertEqual(y.original_tensor.shape, (3, i))
+
+        torch._dynamo.config.error_on_recompile = False
+
+    def test_bucketize(self):
+        """
+        Test that we compile a new graph on a shape that is larger than the original bucket.
+        """
+
+        def f(a, b, c):
+            return a @ b + c
+
+        f = torch.compile(f, fullgraph=True)
+        multipliers = {0: 16, 1: 16}
+
+        for i in range(3, 22):
+            # Every multiple of 16, we allow recompilation.
+            if i % 16 == 0:
+                torch._dynamo.config.error_on_recompile = False
+            # It takes 2 iterations to trace graph with symbolic shapes. So after 2 iterations
+            # after multiples of 16, we disallow recompilation.
+            if i % 16 == 2:
+                torch._dynamo.config.error_on_recompile = True
+
+            a = PaddedTensor.from_tensor(torch.randn([3, 5]), multipliers)
+            b = PaddedTensor.from_tensor(torch.randn([5, i]), multipliers)
+            c = PaddedTensor.from_tensor(torch.randn([3, i]), multipliers)
+
+            y = f(a, b, c)
+
+            self.assertEqual(y.shape, (16, 16 if i <= 16 else 32))
+            self.assertEqual(y.original_tensor.shape, (3, i))
+
+        torch._dynamo.config.error_on_recompile = False
+
+
+class NNOpTests(TestCase):
+    def setUp(self):
+        super().setUp()
+
+    def test_linear_3d(self):
+        """
+        Test linear layer op with 3d input.
+        """
+
+        class TestModule(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.linear = torch.nn.Linear(7, 9)
+
+            def forward(self, x):
+                return self.linear(x)
+
+        mod = TestModule()
+        mod = torch.compile(mod, fullgraph=True)
+
+        multipliers = {0: 16, 1: 16}
+
+        for i in range(3, 9):
+            x = torch.randn((i, i, 7))
+            x_p = PaddedTensor.from_tensor(x, multipliers)
+
+            y = mod(x)
+            y_p = mod(x_p)
+
+            self.assertEqual(y_p.shape, (16, 16, 9))
+            self.assertEqual(y_p.original_tensor.shape, (i, i, 9))
+            self.assertEqual(y, y_p.unpad())
+
+
+class ModelTests(TestCase):
+    def setUp(self):
+        super().setUp()
+
+    def test_transformer(self):
```

Comment:
Hi， may I suggest we mark this test case as `@requires_cuda` since it use cuda specific API `with torch.device("cuda"):`, `orch._dynamo.config.error_on_recompile`.